disco_aws_automation/disco_elasticsearch_archive.py

```diff
@@ -207,34 +225,39 @@ def _archivable_indices(self):
         yesterday = datetime.date.today() - datetime.timedelta(days=1)
         yesterday_str = yesterday.strftime('%Y.%m.%d')
 
-        return [index[0]
-                for index in self._get_all_indices_and_sizes()
-                if index[0][-10:] < yesterday_str]
+        return [index['index']
+                for index in self._get_all_indices_stats()
+                if index['index'][-10:] < yesterday_str]
 
     def _all_index_names(self):
         """ Return list of all indices """
-        return [index[0]
-                for index in self._get_all_indices_and_sizes()]
+        return [index['index']
+                for index in self._get_all_indices_stats()]
 
-    def _get_all_indices_and_sizes(self):
+    def _get_all_indices_stats(self):
         """
-        Return list of all the indices and their sizes sorted by the date from
-        oldest to latest
+        Return list of all the indices, their sizes and their numbers of shards
+        sorted by the date from oldest to latest
         """
-        stats = self.es_client.indices.stats()
+        cat_stats = self.es_client.cat.indices(bytes='b', h='index,store.size,pri,rep')
+        indices_stats = []
+        for stats_line in cat_stats.split('\n'):
+            if stats_line:
+                stats = stats_line.split()
+                indices_stats.append({'index': str(stats[0]),
+                                      'size': int(stats[1]),
+                                      'shards': int(stats[2]) * (int(stats[3]) + 1)})
```

Comment:
stats[2] and stats[3] correspond to `pri` and `rep`, which are the number of primary shards and the number of replicated copies, respectively.